src/core/component.js

```diff
@@ -34,6 +34,7 @@ var Component = module.exports.Component = function (el, attrValue, id) {
   this.el = el;
   this.id = id;
   this.attrName = this.name + (id ? '__' + id : '');
+  this.el.components[this.attrName] = this;
```

Comment:
Why the move here? It seems the entity should be managing the lifecycle of the component when we can, not both ways.